compliance_checker/cf/cf.py

```diff
@@ -2719,44 +2719,95 @@ def check_cell_methods(self, ds):
-
             for match in regex.finditer(psep, method):
-                if match.group('brace') is not None:
-                    valid_modifier.assert_true(match.group('brace_wunit') in {'interval', 'comment', 'area'},
-                                               '{}:cell_methods contains an invalid modifier: {}. It should be one '
-                                               'of interval, comment or area.'
-                                               ''.format(var.name, match.group('brace_wunit')))
-                    if match.group('brace_wunit') == 'interval':
-                        # attempt to get the number for the interval
-                        valid_modifier.out_of += 2
-                        try:
-                            float(match.group('interval_val'))
-                        except ValueError:
-                            valid_modifier.messages.append('{}:cell_methods contains an interval value that does not parse as a numeric value: "{}".'.format(var.name, match.group('interval_val')))
-                        else:
-                            valid_modifier.score += 1
-
-                        # then the units
-                        try:
-                            Unit(match.group('interval_unit'))
-                        except ValueError:
-                            valid_modifier.messages.append('{}:cell_methods interval units "{}" is not parsable by UDUNITS.'.format(var.name, match.group('interval_unit')))
-                        else:
-                            valid_modifier.score += 1
-
-            if valid_modifier.out_of > 0:
-                ret_val.append(valid_modifier.to_result())
+                if match.group('paren_contents') is not None:
+                    # split along spaces followed by words with a colon
+                    # not sure what to do if a comment contains a colon!
+                    ret_val.append(self._check_cell_methods_paren_info(match.group('paren_contents'), var).to_result())
 
         return ret_val
 
+    def _check_cell_methods_paren_info(self, paren_contents, var):
+        """
+        Checks that the spacing and/or comment info contained inside the
+        parentheses in cell_methods is well-formed
+        """
+        #valid_info = TestCtx(BaseCheck.MEDIUM,
+        #                            '§7.3.2 {} has valid cell_methods spacing/other info'.format(var.name))
+        valid_info = TestCtx(BaseCheck.MEDIUM,
+                             '§7.3.3 {} has valid cell_methods modifiers'.format(var.name))
+        # if there are no colons, this is a simple comment
+        # TODO: are empty comments considered valid?
+        if ':' not in paren_contents:
+            valid_info.out_of += 1
+            valid_info.score += 1
+            return valid_info
+        # otherwise, split into k/v pairs
+        kv_pair_pat = r'(\S+:)\s+(.*(?=\s+\w+:)|[^:]+$)\s*'
```

Comment:
Not having quotation marks or commas like many key value pair specifications made this rather tricky to parse.  Hopefully this is robust.